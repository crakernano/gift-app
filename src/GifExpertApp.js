import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ({defaultCategories = []}) => {


    const[categorias, setCategorias] = useState(defaultCategories);


    return(
        <div>
            <h2>GifExpertApp</h2>
            <AddCategory setCategorias={setCategorias}/>
            <hr></hr>

            <ol>
                {
                categorias.map(categoria=> <GifGrid
                                                    category={categoria}
                                                    key={categoria}
                                                    />
                                )
                }

            </ol>
        </div>
    )
}

