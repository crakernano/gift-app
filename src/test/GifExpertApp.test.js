import React from 'react'
import { shallow } from 'enzyme';
import {GifExpertApp} from '../GifExpertApp'


describe('Pruebas en <AddCategory />', ()=>{

    //const wrapper = shallow(<GifExpertApp />);

    test('Debe mostrarse correctamente', ()=>{
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrarse correctamente', ()=>{
        const categorias = ["Pokemon", "Harry Potter"];
        const wrapper = shallow(<GifExpertApp defaultCategories={categorias}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categorias.length);

    });
})