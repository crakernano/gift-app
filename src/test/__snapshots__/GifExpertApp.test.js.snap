// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> Debe mostrarse correctamente 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <ol />
</div>
`;

exports[`Pruebas en <AddCategory /> Debe mostrarse correctamente 2`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Pokemon"
      key="Pokemon"
    />
    <GifGrid
      category="Harry Potter"
      key="Harry Potter"
    />
  </ol>
</div>
`;
